typescript/sdk/src/ism/EvmIsmModule.ts

```diff
@@ -86,4 +150,541 @@ export class EvmIsmModule extends HyperlaneModule<
       config,
     });
   }
+
+  protected async updateRoutingIsm(params: {
+    destination: ChainName;
+    config: RoutingIsmConfig;
+    origin?: ChainName;
+    mailbox?: Address;
+    existingIsmAddress: Address;
+    logger: Logger;
+  }): Promise<IRoutingIsm> {
+    const { destination, config, mailbox, existingIsmAddress, logger } = params;
+    const overrides = this.multiProvider.getTransactionOverrides(destination);
+    let routingIsm: DomainRoutingIsm | DefaultFallbackRoutingIsm;
+
+    // filtering out domains which are not part of the multiprovider
+    config.domains = objFilter(
+      config.domains,
+      (domain, config): config is IsmConfig => {
+        const domainId = this.multiProvider.tryGetDomainId(domain);
+        if (domainId === null) {
+          logger.warn(
+            `Domain ${domain} doesn't have chain metadata provided, skipping ...`,
+          );
+        }
+        return domainId !== null;
+      },
+    );
+
+    const safeConfigDomains = Object.keys(config.domains).map((domain) =>
+      this.multiProvider.getDomainId(domain),
+    );
+
+    const delta: RoutingIsmDelta = existingIsmAddress
+      ? await routingModuleDelta(
+          destination,
+          existingIsmAddress,
+          config,
+          this.multiProvider,
+          this.args.addresses,
+          mailbox,
+        )
+      : {
+          domainsToUnenroll: [],
+          domainsToEnroll: safeConfigDomains,
+        };
+
+    const signer = this.multiProvider.getSigner(destination);
+    const provider = this.multiProvider.getProvider(destination);
+    const owner = await DomainRoutingIsm__factory.connect(
+      existingIsmAddress,
+      provider,
+    ).owner();
+    const isOwner = eqAddress(await signer.getAddress(), owner);
+
+    // reconfiguring existing routing ISM
+    if (existingIsmAddress && isOwner && !delta.mailbox) {
```

Comment:
Can we break up this big fn? Maybe by splitting off these if/else clause contents?